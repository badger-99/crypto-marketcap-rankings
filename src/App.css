h1 {
  color: navy;
}

.column {
  display: flex;
  flex-direction: column;
}

.row {
  display: flex;
  flex-direction: row;
}

.link {
  text-decoration: none;
}

li {
  list-style: none;
}

main {
  border: 1px solid black;
  max-width: 428px;
  height: 100%;
  margin: auto;
}
