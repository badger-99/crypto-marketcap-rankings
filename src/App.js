import { Routes, Route } from 'react-router';
import './App.css';
import Navigation from './components/Navigation';

function App() {
  return (
    <main>
      <h1>Top-10 Crypto Currencies</h1>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/#1" element={<Rank1 />} />
        <Route path="/#2" element={<Rank2 />} />
        <Route path="/#3" element={<Rank3 />} />
        <Route path="/#4" element={<Rank4 />} />
        <Route path="/#5" element={<Rank5 />} />
        <Route path="/#6" element={<Rank6 />} />
        <Route path="/#7" element={<Rank7 />} />
        <Route path="/#8" element={<Rank8 />} />
        <Route path="/#9" element={<Rank9 />} />
        <Route path="/#10" element={<Rank10 />} />
      </Routes>
      <Navigation />
    </main>
  );
}

export default App;
