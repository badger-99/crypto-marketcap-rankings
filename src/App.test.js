import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import App from './App';

const mockStore = configureMockStore([thunk]);

test('App component renders as expected', () => {
  const initialState = {
    crypto: {
      cryptoArray: [
        {
          name: 'testCoin',
          rank: '1',
        },
        {
          name: 'subCoin',
          rank: '2',
        },
        {
          name: 'miniCoin',
          rank: '3',
        },
      ],
    },
  };

  const store = mockStore(initialState);

  const app = render(
    <MemoryRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>,
  );

  expect(app).toMatchSnapshot();
});
