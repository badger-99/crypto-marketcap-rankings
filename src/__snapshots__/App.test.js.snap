// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <div>
          <h1>
            Top-10 Crypto Currencies
          </h1>
          <nav>
            <form>
              <input
                placeholder="Search by Rank"
                type="text"
              />
              <button
                type="submit"
              >
                Search
              </button>
            </form>
            <ul>
              <li>
                <a
                  class="link"
                  href="/1"
                >
                  #1 testCoin
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/2"
                >
                  #2 subCoin
                </a>
              </li>
              <li>
                <a
                  class="link"
                  href="/3"
                >
                  #3 miniCoin
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <div>
        <h1>
          Top-10 Crypto Currencies
        </h1>
        <nav>
          <form>
            <input
              placeholder="Search by Rank"
              type="text"
            />
            <button
              type="submit"
            >
              Search
            </button>
          </form>
          <ul>
            <li>
              <a
                class="link"
                href="/1"
              >
                #1 testCoin
              </a>
            </li>
            <li>
              <a
                class="link"
                href="/2"
              >
                #2 subCoin
              </a>
            </li>
            <li>
              <a
                class="link"
                href="/3"
              >
                #3 miniCoin
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
